items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -92,17 +97,53 @@ class SierraItemUpdater(sierraSource: SierraSource)(
 
     } yield accessConditions
 
-  private def getAvailableDates: Option[List[AvailabilitySlot]] =
-    // call Prismic to get opening times
-    // generate list of dates on now + opening times
-    Some(
-      List(
-        AvailabilitySlot(
-          "2022-01-10T10:00:00+0000",
-          "2022-01-11T10:00:00+0000"
-        )
-      )
+  private def setAvailableDates(
+    item: DisplayItem
+  ): Future[DisplayItem] = {
+    // there is only ever one location per physicalItem and one accessCondition per location,
+    // but this may sometimes be empty if it could not be fetched.
+    val accessCondition = item.locations.headOption
+      .flatMap(_.accessConditions.headOption)
+
+    if ((accessCondition.isDefined && accessCondition.head.isRequestable)) {
+      getAvailableDates(accessCondition.head)
+        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+    } else {
+      Future.successful(item)
+    }
+  }
+
+  private def getAvailableDates(
+    accessCondition: DisplayAccessCondition
+  ): Future[List[AvailabilitySlot]] = {
+    val venueFromDisplayAccessCondition = Map(
+      "online-request" -> "library"
+      // other venue to be added as DisplayAccessMethod id -> content-api venue title
     )
+    val hourNow = clock.getHour
+    val leadTime = accessCondition.method.id match {
+      case "online-request" if hourNow < 10  => 1
+      case "online-request" if hourNow >= 10 => 2
+    }
```

Comment:
This is a tricky one - the `leadTime` here isn't a time, it's a number of availability slots that we're going to `drop`. We should probably rename it something like `leadTimeDays`, but how does the items API know that an availability slot is 1 day? Should it be checking that dropping this number of slots corresponds to a lead time of this number of days? See comment below for a suggestion!